pinc/misc.inc

```diff
@@ -1272,6 +1272,38 @@ function factor_strings($strings)
     return [$left_common, $middles, $right_common];
 }
 
+//--------------------------------------------------------------------------
+
+/**
+ * Given a function and arguments that will generate data, attempt
+ * to load the data from memcached first and fall back to the function and
+ * cache the response. If memcache is not running, this will call the requested
+ * function and return its data without generating a notice/warning/error.
+ *
+ * If the requested function returns data that has localized strings, it's
+ * important that $key_salt = get_desired_language();
+ */
+function memoize_function($function, $args = [], $expire_from_now = 3600, $key_salt = "")
+{
+    $memcache = new Memcached();
+    $memcache->addServer('localhost', 11211);
+
+    $key = hash("sha256", $function . serialize($args) . $key_salt);
```

Comment:
sha256 is resistant to collision attacks, this is a property we probably don't need for caching. Should we instead use something faster?

Reply:
I changed it from md5 to sha256 to reduce the collision space since this has no control over the function name, args, or salt. I don't know what the odds are of a collision with md5 though, so it might not matter and md5 might be good enough.